### PR TITLE
[Feature] Refactors print document feature

### LINKED ISSUE #6588: ✨ Improve readability of printed profiles
## ✨ Feature

Current printed profiles seem to have a big problem in terms of readability, the visual hierarchy is cumbersome and extremely hard to parse for any useful information. We want to improve the quality of the Printed profiles by adding a little formatting to the output. 

## 🕵️ Details

[This is the current state of the printout.](https://drive.google.com/file/d/1rS-9NlK49xb8_hKmcH-npx2M7vpyp-9x/view?usp=sharing)

[This is the proposed formatting.](https://docs.google.com/document/d/13epIYbnfFdYc-T1_d8SqCKhaJcAVJEXDYkhGBmzLLdA/edit?usp=share_link)

In the images below you can see a quick comparison between formatting. If we remove some of the excess copy, exchange the default font, and use a more sensible use of bolding and titles we can reduce much of the visual noise.

## 🎨 Design File

Include a link to the design file (if it exists).

### OLD EXPERIENCE FORMATTING

image

### NEW EXPERIENCE FORMATTING

image

## 🙋‍♀️ Proposed Implementation

(optional) Do you have a proposed implementation?

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] New fonts match design for the profile downloads
- [ ] New heading hierarchy should match designs
- [ ] Experiences shown as a list instead of headings
- [ ] Needs to work with single and multiple profile downloads
- [ ] Check to make sure it works with edge

### PR DESCRIPTION
**Pull request by mnigh**
🤖 Resolves #6588.

## 👋 Introduction

This PR refactors `UserProfileDocument`, `PoolCandidateDocument` and  consolidates them into a single `ProfileDocument` as well as makes the format of the output more readable.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Navigate to `/en/admin/users/:user-id/profile
2. Click **Print Profile** button
3. Navigate to `/en/admin/users`
4. Click **Download Profiles** button
5. Navigate to `/en/admin/pools/:pool-id/pool-candidates
6. Click **Download Profiles** button

## 📁 Files

1. [Candidate profiles_pool.pdf](https://github.com/GCTC-NTGC/gc-digital-talent/files/11710162/Candidate.profiles_pool.pdf)
2. [Candidate profiles.pdf](https://github.com/GCTC-NTGC/gc-digital-talent/files/11710163/Candidate.profiles.pdf)
7. [Candidate profile.pdf](https://github.com/GCTC-NTGC/gc-digital-talent/files/11710164/Candidate.profile.pdf)

## 📸 Screenshots

Screen Shot 2023-06-09 at 17 16 29
Screen Shot 2023-06-09 at 17 16 39
Screen Shot 2023-06-09 at 17 16 49

### CODE REVIEW

File: apps/web/src/components/UserProfile/PrintExperienceByType/ExperienceItem.tsx
Comment (petertgiles) [inline apps/web/src/components/UserProfile/PrintExperienceByType/ExperienceItem.tsx:252]: Intentional that the fragment was turned to a div?

**Review by petertgiles — APPROVED** [review]
Looks great!  I suggest you get a signoff from Jerbo before merging though.

**Comment by Jerryescandon** [comment]
Can we please remove the NOTES section from prints. It is for sure happening on the batch print where we have multiple profiles.
image

**Review by Jerryescandon — APPROVED** [review]
Looks good! 👍

**Pull request merged**
